packages/client/components/SpotlightModal.tsx

```diff
@@ -0,0 +1,209 @@
+  animation: `${fadeUp.toString()} 300ms ${DECELERATE} 300ms forwards`,
+  background: '#FFFF',
+  borderRadius: 8,
+  boxShadow: Elevation.Z8,
+  display: 'flex',
+  flexWrap: 'wrap',
+  height: '80vh',
+  justifyContent: 'center',
+  opacity: 0,
+  overflow: 'hidden',
+  width: isDesktop ? '80vw' : '90vw',
+  zIndex: ZIndex.SPOTLIGHT_MODAL
+}))
+
+const SelectedReflectionSection = styled('div')({
+  alignItems: 'flex-start',
+  background: PALETTE.SLATE_100,
+  borderRadius: '8px 8px 0px 0px',
+  display: 'flex',
+  flexWrap: 'wrap',
+  height: `${SELECTED_HEIGHT_PERC}%`,
+  justifyContent: 'center',
+  padding: 16,
+  position: 'relative',
+  width: '100%'
+})
+
+const SimilarReflectionGroups = styled('div')({
+  display: 'flex',
+  justifyContent: 'center',
+  height: `${SELECTED_HEIGHT_PERC * 2}%`,
+  padding: 16
+})
+
+const Title = styled('div')({
+  color: PALETTE.SLATE_800,
+  fontSize: 16,
+  fontWeight: 600,
+  textAlign: 'center'
+})
+
+const TopRow = styled('div')({
+  width: `calc(100% - 48px)`, // 48px accounts for icon size
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'center'
+})
+
+const ReflectionWrapper = styled('div')({
+  display: 'flex',
+  alignItems: 'center',
+  position: 'absolute',
+  top: `calc(${SELECTED_HEIGHT_PERC / 2}% - ${ElementHeight.REFLECTION_CARD / 2}px)`,
+  left: `calc(50% - ${ElementWidth.REFLECTION_CARD / 2}px)`,
+  zIndex: ZIndex.REFLECTION_IN_FLIGHT_LOCAL
+})
+
+const SearchInput = styled('input')({
+  appearance: 'none',
+  border: `1px solid ${PALETTE.SKY_500}`,
+  borderRadius: 4,
+  boxShadow: `0 0 1px 1px ${PALETTE.SKY_300}`,
+  color: PALETTE.SLATE_700,
+  display: 'block',
+  fontSize: 14,
+  lineHeight: '24px',
+  outline: 'none',
+  padding: '6px 0 6px 39px',
+  width: '100%',
+  '::placeholder': {
+    color: PALETTE.SLATE_600
+  }
+})
+
+const SearchItem = styled(MenuItemLabel)({
+  overflow: 'visible',
+  padding: 0,
+  position: 'absolute',
+  bottom: -16,
+  width: ElementWidth.REFLECTION_CARD
+})
+
+const StyledMenuItemIcon = styled(MenuItemComponentAvatar)({
+  position: 'absolute',
+  left: 8,
+  top: 8
+})
+
+const SearchIcon = styled(Icon)({
+  color: PALETTE.SLATE_600,
+  fontSize: ICON_SIZE.MD24
+})
+
+const StyledCloseButton = styled(PlainButton)({
+  height: 24,
+  position: 'absolute',
+  right: 16
+})
+
+const CloseIcon = styled(Icon)({
+  color: PALETTE.SLATE_600,
+  cursor: 'pointer',
+  fontSize: ICON_SIZE.MD24,
+  '&:hover,:focus': {
+    color: PALETTE.SLATE_800
+  }
+})
+
+interface Props {
+  closeSpotlight: () => void
+  meeting: SpotlightModal_meeting
+  flipRef: (instance: HTMLDivElement) => void
+}
+
+const SpotlightModal = (props: Props) => {
+  const {closeSpotlight, meeting, flipRef} = props
+  const {spotlightReflection} = meeting
+  const isDesktop = useBreakpoint(Breakpoint.NEW_MEETING_SELECTOR)
+  const handleKeyDown = (e: KeyboardEvent<HTMLDivElement>) => {
+    if (e.key === 'Escape') closeSpotlight()
```

Comment:
ahhh, i'm the reason this is like it is.

the reason `onClose` in the parent doesn't fire is because the input field is currently active & we normally don't want to close modals where that is true: https://github.com/ParabolInc/parabol/blob/3abf58205fdebba284cc699f947807efed8b30aa/packages/client/hooks/usePortal.tsx#L86. A good example is when the expanded reflection stack is open & you're editing a card in it. when you hit escape, you want to finish editing that card, not close the whole stack.

In this case, we _do_ want to close no matter what.
maybe `usePortal` should also accept a `closeIgnoresFocus` boolean? that'd save us from having to introduce another keydown handler. whatcha think?

Reply:
an alternative would be to put a keydown handler on the input element that would call blur() on escape. that way we could pick & choose which inputs could make it close. :shrug:

Reply:
Ahh interesting! 

If the user hasn't typed anything into the input and they hit escape, I think they prob want to close the modal as it's unlikely that you'd accidentally hit escape as the first key when typing.

If the user has started typing, is still focused on the input and they hit escape, it's possible that it was an accident or they want to close the modal. I think adding an onBlur works well in this instance because they'd have to hit escape a second time to close it.

In usePortal, I'm now checking the length of the element's value to determine whether we should blur or close it